src/screens/ui/ActivityAutoCompletionScreen.tsx

```diff
@@ -0,0 +1,170 @@
+import {
+  Box,
+  BoxProps,
+  CheckIcon,
+  ImageBackground,
+  Spinner,
+  SubmitButton,
+  Text,
+  TimerSandIcon,
+} from '@shared/ui';
+import { useAutoCompletion } from '@widgets/survey/model';
+
+type Props = {
+  onPressDone?: () => void;
+};
+
+type FlexContainerProps = PropsWithChildren<BoxProps>;
+
+const FlexContainer: FC<FlexContainerProps> = ({ children, ...boxProps }) => {
+  return (
+    <YStack flex={1} alignItems="center" gap={20} {...boxProps}>
+      {children}
+    </YStack>
+  );
+};
+
+const MainContainer: FC<PropsWithChildren> = ({ children }) => {
+  return (
+    <ImageBackground>
+      <YStack flex={1} px={20} gap={20}>
+        {children}
+      </YStack>
+    </ImageBackground>
+  );
+};
+
+const TimesUpScreen: FC<Props> = ({ onPressDone }) => {
+  const { t } = useTranslation();
+
+  return (
+    <>
+      <FlexContainer justifyContent="flex-end">
+        <TimerSandIcon color="#000" size={80} />
+        <Text fontSize={25}>{t('autocompletion:time_limit_reached')}</Text>
+      </FlexContainer>
+      <FlexContainer justifyContent="flex-start">
+        <Text textAlign="center">
+          {t('autocompletion:time_expired', {
+            activityNames: '[First Activity, Second Activity]',
+          })}
+        </Text>
+
+        <Box w="100%">
+          <SubmitButton
+            accessibilityLabel="autocompletion-times-up"
+            mode="dark"
+            onPress={onPressDone}
+            borderRadius={20}
+          >
+            {t('autocompletion:done')}
+          </SubmitButton>
+        </Box>
+      </FlexContainer>
+    </>
+  );
+};
+
+const ProcessingAnswersScreen: FC = () => {
+  const { t } = useTranslation();
+
+  return (
+    <>
+      <FlexContainer justifyContent="flex-end">
+        <Spinner size={80} />
+        <Text fontSize={25}>{t('autocompletion:processing_answers')}</Text>
+      </FlexContainer>
+
+      <FlexContainer justifyContent="flex-start">
+        <Text textAlign="center">{t('autocompletion:preparing_answers')}</Text>
+      </FlexContainer>
+    </>
+  );
+};
+
+const AnswersSubmittedScreen: FC<Props> = ({ onPressDone }) => {
+  const { t } = useTranslation();
+
+  return (
+    <>
+      <FlexContainer justifyContent="flex-end">
+        <Box bg="$alertSuccessIcon" borderRadius={100} p={10}>
+          <CheckIcon color="white" size={60} />
+        </Box>
+        <Text fontSize={25}>{t('autocompletion:answers_submitted')}</Text>
+      </FlexContainer>
+      <FlexContainer justifyContent="flex-start">
+        <Box w="100%">
+          <SubmitButton
+            accessibilityLabel="autocomplete-answers-submitted"
+            mode="dark"
+            onPress={onPressDone}
+            borderRadius={20}
+          >
+            {t('autocompletion:done')}
+          </SubmitButton>
+        </Box>
+      </FlexContainer>
+    </>
+  );
+};
+
+const ActivityAutoCompletionScreen: FC = () => {
```

Comment:
Rename to AutoCompletionScreen. Move the content and logic into separated component, place it into the widgets/survey/ui/autocompletion level like the other components above